### DISCUSSION TITLE
[Bug] Windows system always show non root privileges

### DISCUSSION THREAD

**Discussion by LywLover** [projectdiscovery/naabu#230] [Q&A] [answered]
**Describe the bug**
I run is as administrator on windows system but always got "Running CONNECT scan with non root privileges".

After read the source code I found this function.
https://github.com/projectdiscovery/naabu/blob/b6bca90820180d35fab501038719682079b761a8/v2/pkg/runner/util.go#L13-L15

Getuid returns the numeric user id of the caller, but on Windows, it returns -1 .

**Comment by ehsandeep** [comment, marked as answer]
@LywLover Yes, this is expected, on windows connect scan is the only option to use.

**Comment by LywLover** [comment]
Thanks for you reply. Is there a reason why windows do not support SYN scan?

**Reply by ehsandeep** [reply]
@LywLover the library to include during the build process on windows is not license compatible, so we will try to include instructions to manually build naabu on windows to use SYN scan.

**Reply by vinirossa** [reply]
> @LywLover the library to include during the build process on windows is not license compatible, so we will try to include instructions to manually build naabu on windows to use SYN scan.

There is an instruction already for this? It would help me a lot.
